lib/parser/lexer/literal.rb

```diff
@@ -78,7 +78,7 @@ def initialize(lexer, str_type, delimiter, str_s, heredoc_e = nil,
                       !heredoc?)
 
       # Capture opening delimiter in percent-literals.
-      @str_type << delimiter if @str_type.start_with?('%'.freeze)
+      @str_type += delimiter if @str_type.start_with?('%'.freeze)
```

Comment:
To be more precise: I have no objections to merging this chunk, only the other one.